lib/iris/cube.py

```diff
@@ -781,6 +781,27 @@ def __init__(self, data, standard_name=None, long_name=None,
             for cell_measure, dims in cell_measures_and_dims:
                 self.add_cell_measure(cell_measure, dims)
 
+        # When True indexing may result in a view onto the original data array,
+        # to avoid unnecessary copying.
+        self._share_data = False
+
+    @property
+    def share_data(self):
```

Comment:
If we were to even consider sharing data, I would have thought that using `iris.FUTURE` might also have been an appropriate way to go. It's less intrusive.

Is there a direct need to have this as a property to fit the use case?